pkg/core/consensus/initiator/initiator.go

```diff
@@ -0,0 +1,77 @@
+package initiator
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"github.com/dusk-network/dusk-blockchain/pkg/config"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus/factory"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus/generation"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus/msg"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire"
+	"github.com/dusk-network/dusk-blockchain/pkg/wallet"
+	log "github.com/sirupsen/logrus"
+)
+
+var l = log.WithField("process", "consensus initiator")
+
+func LaunchConsensus(eventBroker wire.EventBroker, rpcBus *wire.RPCBus, w *wallet.Wallet) {
+	// TODO: sync first
+	go startProvisioner(eventBroker, rpcBus, w)
+	go startBlockGenerator(eventBroker, rpcBus, w)
+}
+
+func startProvisioner(eventBroker wire.EventBroker, rpcBus *wire.RPCBus, w *wallet.Wallet) {
+	// Setting up the consensus factory
+	f := factory.New(eventBroker, rpcBus, config.ConsensusTimeOut, w.ConsensusKeys())
+	f.StartConsensus()
+
+	// Get starting round
+	blsPubKey := w.ConsensusKeys().BLSPubKeyBytes
+	startingRound := getStartingRound(blsPubKey, eventBroker)
```

Comment:
With reference to #54, is there a specific reason we use "getStartingRound" (the one that waits for acceptedBlock event) to fetch the current chain tip, instead of fetching the tip from the `db.FetchCurrentHeight`?

Reply:
Note that `blsPubKey`  is not anymore used with latest changes.

Reply:
Correct, I just ran into this while testing. Currently added a `GetLastBlock` call with the RPCBus to get our height. If it is 0, we start without waiting as per #54. One more thing I am currently remembering that wasn't yet finished, is for the node to wait to be synced to the tip. This is part of the reason as well why we use `getStartingRound`, the other being, that we could have a slight chance of mis-timing our entry and missing a lot of crucial messages for agreement, getting the node stuck instantly.

Reply:
Also, correct about `blsPubKey`. will be removed in the next commit, thank you for noticing!